File: src/main/java/cz/linkedlist/TileInfoService.java
package cz.linkedlist;

import cz.linkedlist.info.ProductInfo;
import cz.linkedlist.info.TileInfo;
import org.springframework.util.concurrent.ListenableFuture;

import java.util.List;

/**
 * @author Martin Macko <https://github.com/LinkedList>
 */
public interface TileInfoService {
    TileInfo getTileInfo(TileSet tileSet);

    ProductInfo getProductInfo(TileSet tileSet);

    /**
     * Will download TileInfo for specified tileSet
     * @param tileSet
     * @return new instance of TileSet with downloaded TileInfo
     */
    ListenableFuture<TileSet> downTileInfo(final TileSet tileSet);

    ListenableFuture<List<TileSet>> downTileInfo(final List<TileSet> tileSets);
}

File: src/main/kotlin/cz/linkedlist/TileInfoService.kt
package cz.linkedlist

import cz.linkedlist.info.ProductInfo
import cz.linkedlist.info.TileInfo
import org.springframework.util.concurrent.ListenableFuture

/**
 * @author Martin Macko <https:></https:>//github.com/LinkedList>
 */
interface TileInfoService {
    fun getTileInfo(tileSet: TileSet): TileInfo

    fun getProductInfo(tileSet: TileSet): ProductInfo

    /**
     * Will download TileInfo for specified tileSet
     * @param tileSet
     * @return new instance of TileSet with downloaded TileInfo
     */
    fun downTileInfo(tileSet: TileSet): ListenableFuture<TileSet>

    fun downTileInfo(tileSets: Collection<TileSet>): ListenableFuture<List<TileSet>>
}
